### PR TITLE
Fix vfprintf_chk and add other xprintf_chk functions

### PR DESCRIPTION
**Pull request by thomasten**
* Added missing parameter to __vfprintf_chk
* Added other ..printf_chk functions similar to the two existing ones

### CODE REVIEW

File: libc/__fprintf_chk.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+#include <stdarg.h>
+#include <stdio.h>
+
+// GCC sometimes replaces fprintf() calls with __fprintf_chk() calls. In glibc
```
Comment (dthaler) [inline libc/__fprintf_chk.c]: should this code be inside
#ifdef __GNUC__
?

**Comment by radhikaj** [comment]
@thomasten. Thank you for this contribution. Can you please address feedback. We would love to be able to merge this after it gets approved.

**Comment by thomasten** [comment]
> 
> 
> @thomasten. Thank you for this contribution. Can you please address feedback. We would love to be able to merge this after it gets approved.

@radhikaj I think I already addressed all feedback in 1b95835

File: libc/__fprintf_chk.c [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifdef __GNUC__
+
+#include <openenclave/enclave.h>
+#include <stdarg.h>
+#include <stdio.h>
+
+// GCC sometimes replaces fprintf() calls with __fprintf_chk() calls. In glibc
```
Comment (jhand2) [inline libc/__fprintf_chk.c]: What version of GCC is this and in what situations are you seeing these calls generated? If code is compiled with `-nostdlib`, calls to code in glibc shouldn't be linked, right?

Comment (thomasten) [reply libc/__fprintf_chk.c]: I think it happened when adding a static library to my enclave. I guess this is not supposed to be supported and one should recompile such dependencies?

Comment (jhand2) [reply libc/__fprintf_chk.c]: I could go either way. OE provides a standard C library, so I don't know that dependencies should all need to be compiled with nostdlib (or some other combination of compiler arguments). But supporting every internal function for every compiler could get a bit out of hand. I guess if someone wanted to target a different libc implementation in a non-enclave application they would have to recompile the dependencies in the same way.

@anakrish @CodeMonkeyLeet do either of you have an opinion here?

Comment (anakrish) [reply libc/__fprintf_chk.c]: GCC emits calls to these object size checking functions when compiled with _FORTIFY_SOURCE and optimizations https://gcc.gnu.org/onlinedocs/gcc/Object-Size-Checking.html

Secure coding practices recommend compiling with _FORTIFY_SOURCE (and also other flags like -fstack-protector) https://wiki.debian.org/Hardening

See https://github.com/openenclave/openenclave/issues/1600 for a list of secure flags that OE must be built with.

Comment (jhand2) [reply libc/__fprintf_chk.c]: Ok thats fine. For reference, BSD libc just aliases the functions (i.e. forwards __fprintf_chk to fprintf).

So I'm fine with supporting this in OE.

**Comment by radhikaj** [comment]
@thomasten can you squash the commits? I can then merge this?

**Comment by radhikaj** [comment]
Bors r+

**Comment by jhand2** [comment]
Test failure looks like an infra issue unrelated to this PR

bors r+

**Pull request merged**